src/include/duckdb/common/catch_stacktrace.hpp

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include "duckdb/common/exception.hpp"
+
+#ifdef DUCKDB_DEBUG_STACKTRACE
+#define CATCH_STACKTRACE(X) duckdb::Exception::FormatStackTrace(X).c_str()
+#endif
+
+#include "catch.hpp"
```

Comment:
good idea, will do

Reply:
MotherDuck will enable this on its CI, which as I mentioned is a DuckDB release/unittest

notes:

- DUCKDB_DEBUG_STACKTRACE is not implemented for Windows (see src/common/exception.cpp) so do not try to build it on that platform as it will fail compilation

- this will work on release builds for Linux and MacOS (and maybe also in debug mode on x86/gcc/Linux, as I mentioned on MacOS address sanitising causes the backtrace to no longer work). But the chance of getting a crash is I think generally higher with optimised code. So that would also be an argument to prepare a release CI run.